skcosmo/preprocessing/_flexible_scaler.py

```diff
@@ -0,0 +1,600 @@
+    array([[ 0., -2.,  1.],
+           [-3.,  1.,  2.],
+           [ 3.,  1., -3.]])
+    >>> transformer.scale_ * transformer.transform(X) + transformer.mean_
+        array([[ 1., -2.,  2.],
+               [-2.,  1.,  3.],
+               [ 4.,  1., -2.]])
+    >>>
+    """
+
+    def __init__(
+        self, with_mean=True, with_std=True, column_wise=False, rtol=1e-5, atol=0
+    ):
+        """Initialize StandardFlexibleScaler."""
+        self.with_mean = with_mean
+        self.with_std = with_std
+        self.column_wise = column_wise
+        self.rtol = rtol
+        self.atol = atol
+        self.n_samples_seen_ = 0
+
+    def fit(self, X, y=None, sample_weight=None):
+        """Compute mean and scaling to be applied for subsequent normalization.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features)
+            The data used to compute the mean and standard deviation
+            used for later scaling along the features axis.
+
+        y: None
+            Ignored.
+
+        sample_weight: ndarray of shape (n_samples,)
+            Weights for each sample. Sample weighting can be used to center
+            (and scale) data using a weighted mean. Weights are internally
+            normalized before preprocessing.
+
+        Returns
+        -------
+        self : object
+            Fitted scaler.
+        """
+
+        self.n_samples_seen_, self.n_features_ = X.shape
+
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+            sample_weight = sample_weight / np.sum(sample_weight)
+
+        if self.with_mean:
+            self.mean_ = np.average(X, weights=sample_weight, axis=0)
+        else:
+            self.mean_ = np.zeros(self.n_features_)
+
+        self.scale_ = 1.0
+        if self.with_std:
+            X_mean = np.average(X, weights=sample_weight, axis=0)
+            var = np.average((X - X_mean) ** 2, weights=sample_weight, axis=0)
+
+            if self.column_wise:
+                if np.any(var < self.atol + abs(X_mean) * self.rtol):
+                    raise ValueError("Cannot normalize a feature with zero variance")
+                self.scale_ = np.sqrt(var)
+            else:
+                var_sum = var.sum()
+                if var_sum < abs(np.mean(X_mean)) * self.rtol + self.atol:
+                    raise ValueError("Cannot normalize a matrix with zero variance")
+                self.scale_ = np.sqrt(var_sum)
+
+        return self
+
+    def transform(self, X, y=None):
+        """Normalize a vector based on previously computed mean and scaling.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features)
+            The data used to scale along the features axis.
+
+        y: None
+            Ignored.
+
+        Returns
+        -------
+        X_tr : ndarray of shape (n_samples, n_features)
+            Transformed array.
+        """
```

Comment:
Can you add in the `copy` parameter?

Reply:
How do you want to use it?

Reply:
from `sklearn.preprocessing.StandardScaler`:

```

    copy : bool, default=True
        If False, try to avoid a copy and do inplace scaling instead.
        This is not guaranteed to always work inplace; e.g. if the data is
        not a NumPy array or scipy.sparse CSR matrix, a copy may still be
        returned.
```

```
    def transform(self, X, copy=None):
        """Perform standardization by centering and scaling

        Parameters
        ----------
        X : {array-like, sparse matrix of shape (n_samples, n_features)
            The data used to scale along the features axis.
        copy : bool, default=None
            Copy the input X or not.

        Returns
        -------
        X_tr : {ndarray, sparse matrix} of shape (n_samples, n_features)
            Transformed array.
        """
        check_is_fitted(self)

        copy = copy if copy is not None else self.copy
        X = self._validate_data(X, reset=False,
                                accept_sparse='csr', copy=copy,
                                estimator=self, dtype=FLOAT_DTYPES,
                                force_all_finite='allow-nan')
```